src/test/java/io/stargate/sgv2/jsonapi/api/v1/util/DataApiCommandSenderBase.java

```diff
@@ -0,0 +1,73 @@
+package io.stargate.sgv2.jsonapi.api.v1.util;
+
+import static io.stargate.sgv2.jsonapi.api.v1.util.IntegrationTestUtils.getAuthToken;
+import static io.stargate.sgv2.jsonapi.api.v1.util.IntegrationTestUtils.getCassandraPassword;
+import static io.stargate.sgv2.jsonapi.api.v1.util.IntegrationTestUtils.getCassandraUsername;
+
+import io.stargate.sgv2.jsonapi.config.constants.HttpConstants;
+import io.stargate.sgv2.jsonapi.service.embedding.operation.test.CustomITEmbeddingProvider;
+import java.util.Base64;
+import java.util.Map;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+/**
+ * Helper class used for constructing and sending commands to the Data API for testing purposes
+ * (usually integration tests).
+ */
+public abstract class DataApiCommandSenderBase<T extends DataApiCommandSenderBase> {
+  protected final String namespace;
+
+  protected int expectedHttpStatus = 200;
+
+  protected DataApiCommandSenderBase(String namespace) {
+    this.namespace = namespace;
+  }
+
+  /**
+   * Fluent method for setting the expected HTTP status code of the response; default is 200.
+   *
+   * @param expectedHttpStatus Status to expect
+   * @return Type-safe "this" sender for call chaining
+   */
+  public T expectHttpStatus(int expectedHttpStatus) {
+    this.expectedHttpStatus = expectedHttpStatus;
+    return _this();
+  }
+
+  protected T _this() {
+    return (T) this;
+  }
+
+  protected Map<String, ?> getHeaders() {
```

Comment:
Copied from existing IT base class, no change.

Can allow overriding when needed, not added yet as no need.